Inconsistent JVM-target compatibility

I have a problem similar to this Issue(#29).

Whether it occurs or not seems to depend on the kotlin version.
The build seems to succeed if the version of Kotlin is 1.7.X and fail if the version is 1.8.X or higher.

* android/settings.gradle
```android/settings.gradle
plugins {
    ...
//    // -> Build Success
//    id "org.jetbrains.kotlin.android" version "1.7.10" apply false

    // -> Build Failed
    id "org.jetbrains.kotlin.android" version "1.8.22" apply false
}
```

* android/app/build.gradle
```android/app/build.gradle
android {
    ...
    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_17
        targetCompatibility = JavaVersion.VERSION_17
    }

    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_17
    }
}
```

* android/gradle/wrapper/gradle-wrapper.properties
```android/gradle/wrapper/gradle-wrapper.properties
distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-all.zip
```

---
### Build Result
```.bat
Launching lib\main.dart on XXXXXX in debug mode...
Running Gradle task 'assembleDebug'...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':async_preferences:compileDebugKotlin'.
> 'compileDebugJavaWithJavac' task (current target is 1.8) and 'compileDebugKotlin' task (current target is 17) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 28s
Error: Gradle task assembleDebug failed with exit code 1
```

Many people who have updated to Android Studio Ladybug have encountered the following problem, and will probably take action to increase the Gradle version, and while at it, increase the kotlin version.
https://github.com/flutter/flutter/issues/156307
https://github.com/flutter/flutter/issues/156111
Also, since google_mobile_ads: 5.2.0 requires kotlin version 1.8.X or higher, I think many people will encounter this problem given that many people use this library with respect to advertising.
Therefore, we would appreciate it if you could address this issue.

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -2,14 +2,14 @@ group 'com.svprdga.asyncpreferences'
 version '0.1.0'
 
 buildscript {
-    ext.kotlin_version = '1.7.21'
+    ext.kotlin_version = '1.8.22'
     repositories {
         google()
         mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.3.1'
+        classpath 'com.android.tools.build:gradle:8.3.2'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }
@@ -29,6 +29,9 @@ android {
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
     }
     defaultConfig {
         minSdkVersion 16

--- a/android/gradle/wrapper/gradle-wrapper.properties
+++ b/android/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/example/android/gradle/wrapper/gradle-wrapper.properties
+++ b/example/android/gradle/wrapper/gradle-wrapper.properties
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-all.zip

--- a/example/android/settings.gradle
+++ b/example/android/settings.gradle
@@ -18,8 +18,8 @@ pluginManagement {
 
 plugins {
     id "dev.flutter.flutter-plugin-loader" version "1.0.0"
-    id "com.android.application" version "8.2.2" apply false
-    id "org.jetbrains.kotlin.android" version "1.6.21" apply false
+    id "com.android.application" version "8.3.2" apply false
+    id "org.jetbrains.kotlin.android" version "1.8.22" apply false
 }
 
 include ":app"

--- a/example/pubspec.yaml
+++ b/example/pubspec.yaml
@@ -10,7 +10,7 @@ dependencies:
     sdk: flutter
   async_preferences:
     path: ../
-  cupertino_icons: 1.0.5
+  cupertino_icons: ^1.0.8
 
 dev_dependencies:
   flutter_test:

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -5,8 +5,7 @@ homepage: https://davidserrano.io/
 repository: https://github.com/svprdga/async_preferences
 
 environment:
-  sdk: '>=3.1.0 <4.0.0'
-  flutter: '>=3.3.0'
+  sdk: ^3.5.2
 
 dependencies:
   flutter:

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -5,7 +5,8 @@ homepage: https://davidserrano.io/
 repository: https://github.com/svprdga/async_preferences
 
 environment:
-  sdk: ^3.5.2
+  sdk: '>=3.1.0 <4.0.0'
+  flutter: '>=3.3.0'
 
 dependencies:
   flutter:

